src/estimation/include/iDynTree/Estimation/AttitudeQuaternionEKF.h

```diff
@@ -0,0 +1,332 @@
+    const unsigned int output_dimensions_without_magnetometer = 3;     ///< dimension of \f$ \mathbb{R}^3 \f$ accelerometer measurements
+    const unsigned int input_dimensions = 3;                           ///< dimension of \f$ \mathbb{R}^3 \f$ gyroscope measurements
+
+    /**
+     * @struct AttitudeQuaternionEKFParameters Parameters to set up the quaternion EKF
+     * @var AttitudeQuaternionEKFParameters::time_step_in_seconds
+     * discretization time step in seconds, default value: \f$ 0.01 s \f$
+     * @var AttitudeQuaternionEKFParameters::bias_correlation_time_factor
+     * time factor modeling how fast the bias can vary, default value: \f$ 0.01 \f$
+     * @var AttitudeQuaternionEKFParameters::accelerometer_noise_variance
+     * measurement noise covariance \f$ \sigma_{acc}^{2} \f$ of accelerometer measurement, default value: \f$ 0.001 \f$
+     * @var AttitudeQuaternionEKFParameters::magnetometer_noise_variance
+     * measurement noise covariance \f$ \sigma_{mag}^{2} \f$ of magnetometer measurement, default value: \f$ 0.001 \f$
+     * @var AttitudeQuaternionEKFParameters::gyroscope_noise_variance
+     * system noise covariance \f$ \sigma_{gyro}^{2} \f$ of gyroscope measurement, since it is the input to the system, default value: \f$ 0.001 \f$
+     * @var AttitudeQuaternionEKFParameters::gyro_bias_noise_variance
+     * system noise covariance \f$ \sigma_{gyrobias}^{2} \f$ of gyroscope bias estimate, default value: \f$ 0.0001 \f$
+     * @var AttitudeQuaternionEKFParameters::initial_orientation_error_variance
+     * initial state covariance \f$ \sigma_{q_0}^{2} \f$ of orientation, default value: \f$ 10 \f$
+     * @var AttitudeQuaternionEKFParameters::initial_ang_vel_error_variance
+     * initial state covariance \f$ \sigma_{\omega_0}^{2} \f$ of angular velocity, default value: \f$ 10 \f$
+     * @var AttitudeQuaternionEKFParameters::initial_gyro_bias_error_variance
+     * measurement noise covariance \f$ \sigma_{acc}^{2} \f$ of gyroscope bias, default value: \f$ 10 \f$
+     * @var AttitudeQuaternionEKFParameters::use_magenetometer_measurements
+     * flag to enable the use of magnetometer measurement for yaw correction, default value: false
+     */
+    struct AttitudeQuaternionEKFParameters {
+    double time_step_in_seconds{0.01};
+    double bias_correlation_time_factor{0.01};
+
+    // measurement noise - zero mean, and a given variance
+    double accelerometer_noise_variance{0.001};
+    double magnetometer_noise_variance{0.001};
+
+    // process noise - zero mean, and a given covariance
+    double gyroscope_noise_variance{0.001};
+    double gyro_bias_noise_variance{0.0001};
+
+    double initial_orientation_error_variance{10};
+    double initial_ang_vel_error_variance{10};
+    double initial_gyro_bias_error_variance{10};
+
+    bool use_magenetometer_measurements{false};
```

Comment:
```suggestion
    bool use_magnetometer_measurements{false};
```
There are a few similar typos around.